### PR TITLE
SONARPY-1303 add metadata for Python for S6001

### PR DESCRIPTION
**Pull request by maksim-grebeniuk-sonarsource**

### CODE REVIEW

File: rules/S6001/description.adoc [hunk 1]
```diff
@@ -0,0 +1,4 @@
+When a back reference in a regex refers to a capturing group that hasn't been defined yet (or at all), it can never be matched. Named back references throw a ``++PatternSyntaxException++`` in that case; numeric back references fail silently when they can't match, simply making the match fail.
```
Comment (guillaume-dequenne-sonarsource) [inline rules/S6001/description.adoc:1]: I believe `PatternSyntaxException` is a Java exception, we shouldn't refer to it in the Python rule description. We should probably have a Python-specific `description.adoc` which describe the actual Python behavior.

**Review by guillaume-dequenne-sonarsource — CHANGES_REQUESTED** [review]
I think we need a Python-specific description for this rule.

**Review by guillaume-dequenne-sonarsource — APPROVED** [review]
LGTM.

File: rules/S6001/python/rule.adoc [hunk 2]
```diff
@@ -0,0 +1,39 @@
+include::../description.adoc[]
+
+== Noncompliant Code Example
+
+[source,text]
+----
+import re
+pattern = re.compile(r"\1(.)", re.DOTALL) # Noncompliant, group 1 is defined after the back reference
```
[the remark this answers is not in the document]
Comment (maksim-grebeniuk-sonarsource) [reply rules/S6001/python/rule.adoc]: done

**Pull request merged**